### PR TITLE
Kate / DTRA-404 / [Refactoring]: Divide trade-categories.tsx file and cover changes with tests

### PR DESCRIPTION
**Pull request by kate-deriv**
## Changes:

- Refactoring: 
1. decomposition of `trade-categories.tsx`
2. replace `localize()` ---> `<Localize />`
3. sort imports
- Add unit tests

**Note:** tests for Accumulators will be added in another PR

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/trader/src/Assets/Trading/Categories/trade-categories.tsx [hunk 1]
```diff
@@ -1,11 +1,26 @@
 import React from 'react';
```
Comment (kate-deriv) [inline packages/trader/src/Assets/Trading/Categories/trade-categories.tsx:1]: This file includes text description for unsupported in `DTrader` contracts (like `stay, end, asian, reset, callputspread `ant etc.), but existing in `SmartTrader`. Should we leave this text description for future when(if) these contracts will be transferred to `DTrater`? Or I can delete?

File: packages/trader/src/Assets/Trading/Categories/trade-categories.tsx [hunk 2]
```diff
@@ -14,353 +29,53 @@ const TradeCategories = ({ category, onClick }: { category?: string; onClick: ()
...
-                            )}
-                        </Text>
-                        <Text as='p'>
-                            {localize(
-                                'If you select "Allow equals", you win the payout if exit spot is higher than or equal to entry spot for "Rise". Similarly, you win the payout if exit spot is lower than or equal to entry spot for "Fall".'
-                            )}
-                        </Text>
-                    </React.Fragment>
-                );
+                TradeTypeTemplate = <RiseFallTradeDescription />;
```
Comment (kate-deriv) [inline packages/trader/src/Assets/Trading/Categories/trade-categories.tsx:33]: `rise_fall_equal` and `rise_fall` have same text description, so wrote common case for them.

File: packages/trader/src/Assets/Trading/Categories/Description/multiplier-trade-description.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    }
+                />
+            </Text>
+            <Text as='p'>
+                <Localize i18n_default_text={'The exit spot is the market price when the contract is closed.'} />
+            </Text>
+        </React.Fragment>
+    );
+};
+
```
Comment (kate-deriv) [inline packages/trader/src/Assets/Trading/Categories/Description/multiplier-trade-description.tsx:101]: I also tried approach from other files (store all `<Localize />` in array with objects and parse them with the help of `.map `method), but the array with objects become too much multileveled, so I decided to leave file in the same condition as before refactoring.
Example of another approach (regected by me):
```
const content = [
        {
            type: 'paragraph',
            text: (
                <Localize
                    i18n_default_text={
                        'Predict the market direction and select either “Up” or “Down” to open a position. We will charge a commission when you open a position.'
                    }
                />
            ),
        },
        {
            type: 'list',
            text: [
                {
                    type: 'list_item',
                    text: (
                        <Localize
                            i18n_default_text={
                                'If you select “Take profit” and specify an amount that you’d like to earn, your position will be closed automatically when your profit is more than or equals to this amount. Your profit may be more than the amount you entered depending on the market price at closing.'
                            }
                        />
                    ),
                },
                {
                    type: 'list',
                    text: [
                        {
                            type: 'list_item',
                            text: (
                                <Localize
                                    i18n_default_text={
                                        'If you select “Take profit” and specify an amount that you’d like to earn, your position will be closed automatically when your profit is more than or equals to this amount. Your profit may be more than the amount you entered depending on the market price at closing.'
                                    }
                                />
                            ),
                        },
                    ],
                },
            ],
        },
    ] as const;
    const parser = (array: Partial<typeof content>) => {
        return array.map(({ type, text }, index) => {
            if (type === 'paragraph') {
                return (
                    <Text as='p' key={index.toString() + text}>
                        {text}
                    </Text>
                );
            }
            if (type === 'list_item') {
                return (
                    <Text as='li' key={index.toString() + text}>
                        {text}
                    </Text>
                );
            }
            return <ul key={index.toString()}>{parser(text)}</ul>;
        });
    };
    console.log('test', parser(content));
    return <React.Fragment>{parser(content)}</React.Fragment>;

```

File: packages/trader/src/Assets/Trading/Categories/trade-categories.tsx (hunk 1 above)
Comment (maryia-deriv) [reply packages/trader/src/Assets/Trading/Categories/trade-categories.tsx:1]: yes, please keep

File: packages/trader/src/Assets/Trading/Categories/Description/multiplier-trade-description.tsx (hunk 3 above)
Comment (maryia-deriv) [reply packages/trader/src/Assets/Trading/Categories/Description/multiplier-trade-description.tsx:101]: @kate-deriv no need to create 1 multi-leveled object, you can create 4 arrays with content, e.g. array #1 for the first block of paragraphs, array #2 for the block of li's inside the 1st `ul`, array #3 for the block of li's inside the 2nd `ul` (or array #2 and array #3 can be united into an object), array #4 for the last 2 paragraphs. naming is up to you but it's better to make it semantic, e.g.  about_section / multipliers_params_section / deal_cancellation_section / closing_section

**Review by maryia-deriv — CHANGES_REQUESTED** [review]
just a couple of comments, the rest LGTM :)

File: packages/trader/src/Assets/Trading/Categories/Description/vanilla-trade-description.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <Localize
+            i18n_default_text='You may sell the contract up until 60 seconds before expiry. If you do, we’ll pay you the <0>contract value</0>.'
+            components={[<span className='contract-type-info__content-definition' onClick={onClick} key={0} />]}
+            key='5'
+        />,
+    ];
+    return (
+        <React.Fragment>
+            {content.map((paragraph, index) => (
+                <Text as='p' key={index.toString() + paragraph}>
```
[the remark this answers is not in the document]
Comment (kate-deriv) [reply packages/trader/src/Assets/Trading/Categories/Description/vanilla-trade-description.tsx:38]: Will do for all contract types except TurbosTradeDescription and CallPutSpreadTradeDescription as they have no keys as props

File: packages/trader/src/Assets/Trading/Categories/Description/vanilla-trade-description.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Localize } from '@deriv/translations';
+import { Text } from '@deriv/components';
+
+const VanillaTradeDescription = ({ onClick }: { onClick: () => void }) => {
```
[the remark this answers is not in the document]
Comment (kate-deriv) [reply packages/trader/src/Assets/Trading/Categories/Description/vanilla-trade-description.tsx:5]: Will do, thank you

**Review by maryia-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

**Review by henry-deriv — APPROVED** [review]

**Review by akmal-deriv — APPROVED** [review]

**Pull request merged**